src/solvers/nonlinear_solid_solver.cpp

```diff
@@ -124,9 +124,8 @@ void NonlinearSolidSolver::completeSetup()
   for (auto& nat_bc_data : nat_bdr_) {
     SLIC_ASSERT_MSG(std::holds_alternative<std::shared_ptr<mfem::VectorCoefficient>>(nat_bc_data.coef),
                     "Traction boundary condition had a non-vector coefficient.");
-    H_form->AddBdrFaceIntegrator(
-        new HyperelasticTractionIntegrator(*std::get<std::shared_ptr<mfem::VectorCoefficient>>(nat_bc_data.coef)),
-        nat_bc_data.markers);
+    H_form->AddBdrFaceIntegrator(nat_bc_data.newVecIntegrator<HyperelasticTractionIntegrator>().release(),
```

Comment:
Just to clarify, my thinking here was that wrapping up the construction of an object using part of internal class state (which I also did on my WIP FEState refactor) minimizes exposure of internal members (otherwise you'd need to write an get-function) without obfuscating what the code is doing (adding a new integrator based on that boundary condition).

Alternatively, the call could be structured as `bc.addBdrFaceVecIntegrator<HyperelasticTractionIntegrator>(H_form)` which in my opinion still converys the meaning "add a boundary face integrator of type `HyperelasticTractionIntegrator` to` H_form` based on boundard condition `bc`"

Reply:
I think I like that call a bit better.

Reply:
I've been thinking about this, and I think at the end of the day I've settled on the function that you originally developed. It is clear that you are building a new integrator and passing it off to the mfem bilinear form, so I think it wins out.